### PR TITLE
Add ServerComms abstract base class

### PR DESCRIPTION
**Pull request by xen-0**

### CODE REVIEW

File: malcolm/core/servercomms.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class ServerComms(Loggable):
+    """Abstract class for dispatching requests to a process and responses to a
+    client"""
+
+    def __init__(self, name, process, sync_factory):
+        super(ServerComms, self).__init__(logger_name=name)
+        self.process = process
+        self.sync_factory = sync_factory
```
Comment (thomascobb) [inline malcolm/core/servercomms.py]: Can you pick up https://diamondtest.atlassian.net/browse/MALCOLM-45 please?

Comment (xen-0) [inline malcolm/core/servercomms.py]: Do that before merging this in or after?

File: malcolm/core/servercomms.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class ServerComms(Loggable):
+    """Abstract class for dispatching requests to a process and responses to a
+    client"""
+
+    def __init__(self, name, process):
+        super(ServerComms, self).__init__(logger_name=name)
+        self.process = process
+        self.q = self.process.sync_factory.create_queue()
```
Comment (thomascobb) [inline malcolm/core/servercomms.py]: self.process.create_queue()

File: malcolm/core/servercomms.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        raise NotImplementedError(
+            "Abstract method that must be implemented by deriving class")
+
+    def send_to_process(self, request):
+        """Send request to process"""
+        self.process.q.put(request)
+
+    def start(self):
+        """Start communications"""
+        self._send_spawned = self.process.sync_factory.spawn(self.send_loop)
```
Comment (thomascobb) [inline malcolm/core/servercomms.py]: self.process.spawn()

**Pull request merged**